common/flatpak-installation.c

```diff
@@ -2961,67 +3080,46 @@ flatpak_installation_list_unused_refs (FlatpakInstallation *self,
   refs_hash = g_hash_table_new (g_str_hash, g_str_equal);
   refs = g_ptr_array_new_with_free_func (g_object_unref);
 
-  for (i = 0; app_refs[i] != NULL; i++)
+  /* For each app, note the runtime, sdk, and related refs */
+  find_used_refs_for_apps (dir, NULL, app_refs, arch, used_runtimes, used_refs);
+
+  /* If @self is a system installation, also check the per-user installation
+   * for any apps there using runtimes in the system installation or runtimes
+   * there with sdks or extensions in the system installation. Only do so if
+   * the per-user installation exists; it wouldn't make sense to create it here
+   * if not.
```

Comment:
I don't think this part is quite right. We should only mark (e.g.) a system runtime as used if there is an app in the user dir that depends on it, but the matching runtime is not also in the user directory.

Reply:
Yeah, I thought about that, not sure why it didn't make it into the implementation

Reply:
Changed the implementation now